masonry/examples/grid_masonry.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Shows how to use a grid layout in Masonry.
+
+// On Windows platform, don't show a console when opening the app.
+#![windows_subsystem = "windows"]
+
+use masonry::app_driver::{AppDriver, DriverCtx};
+use masonry::dpi::LogicalSize;
+use masonry::widget::{Button, Grid, GridParams, Prose, RootWidget, SizedBox};
+use masonry::{Action, Color, PointerButton, WidgetId};
+use winit::window::Window;
+
+struct Driver {
+    grid_spacing: f64,
+}
+
+impl AppDriver for Driver {
+    fn on_action(&mut self, ctx: &mut DriverCtx<'_>, _widget_id: WidgetId, action: Action) {
+        if let Action::ButtonPressed(button) = action {
+            if button == PointerButton::Primary {
+                self.grid_spacing += 1.0;
+            } else if button == PointerButton::Secondary {
+                self.grid_spacing -= 1.0;
+            } else {
+                self.grid_spacing += 0.5;
+            }
+
+            ctx.get_root::<RootWidget<Grid>>()
+                .get_element()
+                .set_spacing(self.grid_spacing);
+        }
+    }
+}
+
+fn grid_button(params: GridParams) -> Button {
+    Button::new(format!(
+        "X: {}, Y: {}, W: {}, H: {}",
+        params.x, params.y, params.width, params.height
+    ))
+}
+
+pub fn main() {
+    let label = SizedBox::new(
+        Prose::new("Change spacing by right and\n left clicking on the buttons")
+            .with_text_size(14.0),
```

Comment:
Minor bikeshedding:

```suggestion
        Prose::new("Change spacing by right and\nleft clicking on the buttons")
            .with_text_size(14.0),
```

Also, for what it's worth, text should probably be centered.

Reply:
How about:
```suggestion
        Prose::new("Change spacing by right and\n\
                               left clicking on the buttons")
            .with_text_size(14.0),
```